### DISCUSSION TITLE
Catch errors from wasm / button handlers

### DISCUSSION THREAD

**Discussion by markkimsal** [DioxusLabs/dioxus#2316] [Q&A]
If you take the included counter example, change all the counters from i32 to u32, when you try to go "down low" from 0 then the counter simply stops working.  There are no errors in the console.

If you do the overflow math inside the the code, you can see an error, but not when it's in an onclick handler.  I think the error comes from the import handler in js.

```rs
#[component]
fn Home() -> Element {
    let mut count = use_signal(|| 0 as u32);
    // *count.write() -= 1;   // browser console shows underflow error

    rsx! {
        Link {
            to: Route::Blog {
                id: count()
            },
            "Go to blog"
        }
        div {
            h1 { 
                "High-Five counter: {count}" }
            button {
                 onclick: move |_| *count.write() += 1, "Up high!" }
            button { onclick: move |_| *count.write() -= 1, "Down low!" } // no errors, counter stops working, adding stops working
        }
    }
}
```

**Comment by ealmloff** [comment]
WASM doesn't support recovering from panics in rust, but you can get errors to show up in the console with the console_error_panic_hook crate. If you have the default features enabled on dioxus web with the default config, we automatically insert the handler [here](https://github.com/DioxusLabs/dioxus/blob/c9ab09b3480ba2ad3b77d16c5e86382a62258194/packages/web/src/lib.rs#L70-L72). Here is the error message with the default features:

![Screenshot 2024-04-16 at 10 50 46 AM](https://github.com/DioxusLabs/dioxus/assets/66571940/f1c18684-eb32-4ea3-a2f6-ae14718332de)

The first error message points to the right location

**Reply by markkimsal** [reply]
Thanks for that answer.  I do get a console error now after adding features = ["default"] to my cargo.

I have so many more questions though.  Default features are not on by default?  shouldn't `dx new` add features = ["default"] to the cargo?  What other features are available?  It's very hard to search for "default" and "features".

Ok, I removed the default features flag and I still get the error in the console now. I'm really confused beyond the ability to formulate questions.

I found a list of features here (https://docs.rs/crate/dioxus/0.5.1/features), but there's no mention of `panic_hook` which is referenced in your link.
